src/gebt_poc/solver.cpp

```diff
@@ -32,116 +32,126 @@ Kokkos::View<double*> Interpolate(
-Kokkos::View<double**> CalculateSectionalStiffness(
+void CalculateSectionalStiffness(
     const StiffnessMatrix& stiffness, Kokkos::View<double**> rotation_0,
-    Kokkos::View<double**> rotation
+    Kokkos::View<double**> rotation, Kokkos::View<double**> sectional_stiffness
 ) {
-    auto total_rotation = gen_alpha_solver::multiply_matrix_with_matrix(rotation, rotation_0);
+    auto total_rotation = Kokkos::View<double**>(
+        "total_rotation", kNumberOfVectorComponents, kNumberOfVectorComponents
+    );
+    KokkosBlas::gemm("N", "N", 1., rotation, rotation_0, 0., total_rotation);
 
-    // rotation_matrix_6x6 = [total_rotation [0]_3x3; [0]_3x3 total_rotation]
+    // rotation_matrix_6x6 = [
+    //    [total_rotation]          [0]_3x3
+    //        [0]_3x3           total_rotation
+    // ]
     auto rotation_matrix = Kokkos::View<double**>(
         "rotation_matrix", kNumberOfLieGroupComponents, kNumberOfLieGroupComponents
     );
-    Kokkos::parallel_for(
-        Kokkos::MDRangePolicy<Kokkos::DefaultExecutionSpace, Kokkos::Rank<2>>(
-            {0, 0}, {kNumberOfVectorComponents, kNumberOfVectorComponents}
-        ),
-        KOKKOS_LAMBDA(const size_t i, const size_t j) {
-            rotation_matrix(i, j) = total_rotation(i, j);
-            rotation_matrix(i + 3, j + 3) = total_rotation(i, j);
-        }
+    Kokkos::deep_copy(rotation_matrix, 0.);
+    auto rotation_matrix_1 =
+        Kokkos::subview(rotation_matrix, Kokkos::make_pair(0, 3), Kokkos::make_pair(0, 3));
+    Kokkos::deep_copy(rotation_matrix_1, total_rotation);
+    auto rotation_matrix_2 =
+        Kokkos::subview(rotation_matrix, Kokkos::make_pair(3, 6), Kokkos::make_pair(3, 6));
+    Kokkos::deep_copy(rotation_matrix_2, total_rotation);
+
+    // Calculate the sectional stiffness matrix in inertial basis
+    Kokkos::deep_copy(sectional_stiffness, 0.);
+    auto stiffness_matrix_left_rot =
+        Kokkos::View<double**>("temp", kNumberOfLieGroupComponents, kNumberOfLieGroupComponents);
+    KokkosBlas::gemm(
+        "N", "N", 1., rotation_matrix, stiffness.GetStiffnessMatrix(), 0., stiffness_matrix_left_rot
     );
-
-    auto sectional_stiffness = gen_alpha_solver::multiply_matrix_with_matrix(
-        gen_alpha_solver::multiply_matrix_with_matrix(
-            rotation_matrix, stiffness.GetStiffnessMatrix()
-        ),
-        gen_alpha_solver::transpose_matrix(rotation_matrix)
+    KokkosBlas::gemm(
+        "N", "T", 1., stiffness_matrix_left_rot, rotation_matrix, 0., sectional_stiffness
     );
-
-    return sectional_stiffness;
 }
 
-Kokkos::View<double*> CalculateElasticForces(
+void CalculateElasticForces(
     const Kokkos::View<double*> sectional_strain, Kokkos::View<double**> rotation,
     const Kokkos::View<double*> pos_vector_derivatives,
     const Kokkos::View<double*> gen_coords_derivatives,
-    const Kokkos::View<double**> sectional_stiffness
+    const Kokkos::View<double**> sectional_stiffness, Kokkos::View<double*> elastic_forces
```

Comment:
Let's return the elastic forces as 2 views here, since that's how they're both assembled and used.

Reply:
Thanks for pointing out this and the following interface improvement, I had this in mind as well.